src/main/java/org/jenkinsci/plugins/workflow/cps/CpsBodyExecution.java

```diff
@@ -172,29 +179,63 @@ private Continuable createContinuable(CpsThread currentThread, CpsCallableInvoca
     }
 
     @Override
-    public synchronized Collection<StepExecution> getCurrentExecutions() {
+    public Collection<StepExecution> getCurrentExecutions() {
         if (thread==null)   return Collections.emptyList();
-
-        StepExecution s = thread.getStep();
-        if (s!=null)        return Collections.singleton(s);
-        else                return Collections.emptyList();
+        List<StepExecution> executions = new ArrayList<>();
+        // cf. trick in CpsFlowExecution.getCurrentExecutions(true)
+        Map<FlowHead, CpsThread> m = new LinkedHashMap<>();
+        // TODO access to CpsThreadGroup.threads should be restricted to the CPS VM thread, but the API signature does not allow us to return a promise or throw InterruptedException
+        for (CpsThread t : thread.group.threads.values()) {
```

Comment:
Well, yes—see the `TODO` comment right above. Unfortunately when @kohsuke defined this API method, besides implementing it incorrectly here, he did not allow it to throw any exceptions—and to implement it correctly (to run in the CPS VM thread) we would need to either throw exceptions, or return a `ListenableFuture`.

At any rate it appears this method was never actually called, so I am not sure it matters. Perhaps the API method should be deprecated and a better replacement introduced if and when there is a need.